### PR TITLE
wrangler 2: route changes, zone id needed for subdomain zones

### PR DESCRIPTION
**Pull request by e271828-**
These behavior changes are both critical and difficult to catch prior to deployment.

### CODE REVIEW

File: content/workers/wrangler/migration/deprecations.md [hunk 1]
```diff
@@ -29,9 +29,10 @@ Here are common fields that are no longer required.
 
 ## Changes to routes
 
-Please note a breaking change has been introduced to routes: Cloudflare dashboard-defined routes will be **deleted upon deployment** if you have any Wrangler-defined routes.
+Please note a breaking change has been introduced to routes: all Cloudflare dashboard-defined routes for your worker will be **deleted upon deployment** if you have any Wrangler-defined routes for that worker.
 
 - Wrangler-defined routes are the `route` or `routes` key in your `wrangler.toml`. 
+- Cloudflare dashboard-defined routes are any routes you add manually in the dashboard.
 - If both are defined, only routes defined in `wrangler.toml` will be valid. 
 - To manage routes via the Cloudflare dashboard only, update your `wrangler.toml` file as follows: remove any `route` and `routes` keys and add `workers_dev = false`.
```
Comment (deadlypants1973) [inline content/workers/wrangler/migration/deprecations.md]:
```suggestion
- To manage routes via the Cloudflare dashboard only, update your `wrangler.toml` file by removing any `route` and `routes` keys and add `workers_dev = false`.
```

File: content/workers/wrangler/migration/deprecations.md [hunk 2]
```diff
@@ -29,9 +29,10 @@ Here are common fields that are no longer required.
 
 ## Changes to routes
 
-Please note a breaking change has been introduced to routes: Cloudflare dashboard-defined routes will be **deleted upon deployment** if you have any Wrangler-defined routes.
+Please note a breaking change has been introduced to routes: all Cloudflare dashboard-defined routes for your worker will be **deleted upon deployment** if you have any Wrangler-defined routes for that worker.
```
Comment (deadlypants1973) [inline content/workers/wrangler/migration/deprecations.md]:
```suggestion
{{<Aside type="warning" header="Breaking change for routes">}}

All Cloudflare dashboard-defined routes for your Worker will be deleted upon deployment if you have any Wrangler-defined (`wrangler.toml`) routes for that Worker.

{{</Aside>}}
```

**Review by deadlypants1973 — APPROVED** [review]

File: content/workers/wrangler/migration/deprecations.md (hunk 2 above)
Comment (Mehrdadmehrdadian) [reply content/workers/wrangler/migration/deprecations.md:32]: 🙏هنگام بازدید از blockchain.com

کد خطا: 1020

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]
We were not able to reproduce this, given a subdomain we used. Could you please provide a minimal reproduction of a Worker that is doing this.

**Comment by e271828-** [comment]
@JacobMGEvans should be easy to repro. Create a root domain zone and two enterprise subdomain zones. Create some manual routes to workers. Switch to w2 with some routes defined using the subdomain zone hostnames without the ID, and watch everything manual get blown away with nothing added from config on the subdomain zones.

Also, we reported this behavior months ago. Perhaps they've fixed it since then. Have you checked your changelogs for related changes?

**Comment by JacobMGEvans** [comment]
> @JacobMGEvans should be easy to repro. Create a root domain zone and two enterprise subdomain zones. Create some manual routes to workers. Switch to w2 with some routes defined using the subdomain zone hostnames without the ID, and watch everything manual get blown away with nothing added from config on the subdomain zones.
> 
> Also, we reported this behavior months ago. Perhaps they've fixed it since then. Have you checked your changelogs for related changes?

I understand, we still would like to see perhaps a GitHub repo that specifically does this `It can be deduced from the routes directly **unless** you are using subdomain zones, in which case you must specify it explicitly.` We would also likely treat this as a bug, not something to document. 

Which is a separate issue from from the routes in Wrangler configuration blowing away the ones upstream, as for the source of truth issues with publishing routes, we do have the issue in backlog (https://github.com/cloudflare/wrangler2/issues/2082) and @admah is planning out the future work, it is on our radar 😃 .

**Comment by JacobMGEvans** [comment]
Thank you for the PR, we would appreciate  a minimal reproduction of a Worker that is doing this in `workers-sdk` for Wrangler https://github.com/cloudflare/workers-sdk/issues/new?assignees=&labels=bug&template=bug-template.yaml&title=%F0%9F%90%9B+BUG%3A

**Pull request closed without merging**